fragments/k8s/cdk-converged/bundle.yaml

```diff
@@ -0,0 +1,111 @@
+# This is an incomplete bundle fragment. Do not attempt to deploy.
+description: |-
+    A seven-machine Kubernetes cluster, appropriate for bare metal using MaaS KVM Pods. Includes three Kubernetes-masters and four worker nodes.
+series: xenial
+machines:
+  0:
+    constraints: cores=8 mem=16G root-disk=100G
+  1:
+    constraints: cores=8 mem=16G root-disk=100G
+  2:
+    constraints: cores=8 mem=16G root-disk=100G
+  3:
+    constraints: cores=12 mem=32G root-disk=200G
+  4:
+    constraints: cores=12 mem=32G root-disk=200G
+  5:
+    constraints: cores=12 mem=32G root-disk=200G
+  6:
+    constraints: cores=12 mem=32G root-disk=200G
+services:
+  "kubernetes-master":
+    charm: "cs:~containers/kubernetes-master"
+    num_units: 3
+    options:
+      channel: 1.10/stable
+    annotations:
+      "gui-x": "800"
+      "gui-y": "850"
+    to:
+      - 0
+      - 1
+      - 2
+  "kubeapi-load-balancer":
+    charm: "cs:~containers/kubeapi-load-balancer"
+    num_units: 1
+    expose: true
+    annotations:
+      "gui-x": "450"
+      "gui-y": "250"
+    to:
+     - 3
+  "easyrsa":
+    charm: "cs:~containers/easyrsa"
+    num_units: 1
+    annotations:
+      "gui-x": "450"
+      "gui-y": "550"
+    to:
+     - lxd:kubernetes-master
+  "kubernetes-worker":
+    charm: "cs:~containers/kubernetes-worker"
+    constraints: cores=2 mem=4G root-disk=30G
+    num_units: 12
+    options:
+      channel: 1.10/stable
+    expose: true
+    annotations:
+      "gui-x": "100"
+      "gui-y": "850"
+    to: ["kvm:kubernetes-worker-host/0",
+        "kvm:kubernetes-worker-host/1",
+        "kvm:kubernetes-worker-host/2",
+        "kvm:kubernetes-worker-host/3",
+        "kvm:kubernetes-worker-host/0",
+        "kvm:kubernetes-worker-host/1",
+        "kvm:kubernetes-worker-host/2",
+        "kvm:kubernetes-worker-host/3",
+        "kvm:kubernetes-worker-host/0",
+        "kvm:kubernetes-worker-host/1",
+        "kvm:kubernetes-worker-host/2",
+        "kvm:kubernetes-worker-host/3"]
```

Comment:
What's the motive for having 12 worker VMs across 4 physical hosts? My instinct would be to provide 1 worker per physical machine so that Kubernetes can utilize the hardware directly.

Reply:
Utilization is the answer. There is a limit of 110 pods that can be scheduled on a single worker currently and this allows the machine to do more work by effectively allowing more than 110 pods per physical machine.